Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabPublicInterface.h

```diff
@@ -143,6 +144,13 @@ namespace AzToolsFramework
              */
             virtual AZ::IO::Path GetOwningInstancePrefabPath(AZ::EntityId entityId) const = 0;
 
+            /**
+             * Get the template id to the prefab file for the prefab instance owning the entity provided.
+             * @param entityId The id for the entity being queried.
+             * @return Returns the template id to the prefab, or an invalid template id if the entity is owned by the level.
+             */
+            virtual TemplateId GetOwningInstancePrefabTemplateId(AZ::EntityId entityId) const = 0;
```

Comment:
This function does not belong in the PrefabPublicInterface - as per the description:
```
        /*!
         * PrefabPublicInterface
         * Interface to expose Prefab functionality directly to UI and Scripting.
         * Functions will correctly call the Undo/Redo system under the hood.
         */
```

All functions in this API are meant to be used by external classes that don't have direct visibility into the prefab backend (Instance references, Template Ids etc).

Consider moving this call to the PrefabSystemComponentInterface or another backend interface (that does not have Public in its name).

Reply:
I agree with @AMZN-daimini that this function shouldn't be in the public handler. This can be put in PrefabInstanceUtils created by @chiyenteng in another upcoming PR. I suggest that you wait till that gets pushed and add this there.

Reply:
Alternatively, this doesn't even have to be a helper function. In the PrefabSaveHandler, you can directly call FindOwningInstance() and call instance.GetTemplateId()

Reply:
Thanks for all the comments! They helped me understand each component class better.

Beside that, I have a question about the invocation of `m_prefabSaveHandler.ExecuteSavePrefabDialog()` method in the manager.

Since `ContextMenu_SavePrefab` is a static method, I need to do something like this to get `m_prefabSaveHandler`. Wondering if there is a more elegant approach that I might miss.

```cpp
// PrefabIntegrationManager.cpp
AZ::Interface<PrefabIntegrationManager>::Get()->m_prefabSaveHandler.ExecuteSavePrefabDialog(templateId, true);
```

Reply:
I guess either
- We change that function to no longer be static
- We introduce a new interface for PrefabSaveHandler and get that.

(Note that in your code example you should always verify if the return of `AZ::Interface::Get()` is not `nullptr` as the interface may not be handler)